packages/devtools_app/lib/src/split.dart

```diff
@@ -102,13 +108,42 @@ class _SplitState extends State<Split> {
     return LayoutBuilder(builder: _buildLayout);
   }
 
+  @override
+  void dispose() {
+    preferences.splitFractions.removeListener(onChangeSplitFractions);
+    super.dispose();
+  }
+
+  void onChangeSplitFractions() {
+    setState(() {
+      fractionsStorage = jsonDecode(preferences.splitFractions.value);
+    });
+  }
+
   Widget _buildLayout(BuildContext context, BoxConstraints constraints) {
     final width = constraints.maxWidth;
     final height = constraints.maxHeight;
     final axisSize = isHorizontal ? width : height;
 
     final availableSize = axisSize - _totalSplitterSize();
 
+    final routerDelegate = DevToolsRouterDelegate.of(context);
+    String childrenRuntimeType = '';
```

Comment:
this is very clever. I like using the current page as part of the key. I'm not sure about using the types of the children widgets. Instead, I'd prefer to make specifying a name for each splitter required and using that name as part of the key.
In either case, remove the hashCode and just include the entire string in the key.

Reply:
I use list children as storage key because already have many widget use Split and don't want pass 'storageKey' argument to them all.
This way helpful in case Split has dynamic children since Split required children length == fractions length.
So if list children widget of Split change then fractions need change too.

In case save with passed 'storageKey' string, when list children change then saved fractions length can not equal children length (storageKey still same if you not dynamic change), it lead to can not use saved fractions for this Split.
Also resize Split override saved fractions (new and old different length) => when old list children show it will can not use again and will override again more time.
=> If save with 'storageKey' argument you will need handle this argument each time list children widget change for avoid conflict.

Are you think this worthy for use list children name instead of pass storageKey argument? If not I will try make specifying a name for each splitter.

About hashCode, yes your suggest good. Before I use it for 'shorten' string because can have long storage key 'widget1_widget2_widget3_widget4...' but infact Split just have some children, so It good to not use hashCode.

Reply:
I think this change is worthy of landing as long as we switch to using a split names. This is definitely solving a problem users care about fixing.
I'm fine with passing in the # of split children after the split name to ensure there aren't conflicts if the # of children changes.